### PR TITLE
Adding the `asn` field to all `GET` endpoints that return a `server` object

### LINKED ISSUE #7064: All endpoints need to account for the ASN field in servers
## This Bug Report affects these Traffic Control components:

- Traffic Ops

## Current behavior:

After https://github.com/apache/trafficcontrol/pull/7023 was merged, some endpoints that return the represenation of a server in the response don't have the ASN field accounted for. These routes will need a `4.1` api version equivalent, so that these new routes can account for the new field.

## Expected behavior:

All 3.1 routes should return a consistent representation of a server object.

## Steps to reproduce:

Except for the `GET /api/4.1/servers` endpoint, all other 4.1 routes return a server representation that does not have the `asns` field in it. This needs to be fixed so that all routes return the same server object representation.

### PR DESCRIPTION
**Pull request by srijeet0406**
This PR closes #7064 and #7085

<hr/>

## Which Traffic Control components are affected by this PR?

- Traffic Ops

## What is the best way to verify this PR?

Run Traffic Ops locally.
Make requests for any api version other than `4.1` to any endpoint that returns a `server` structure, for example:
`GET deliveryservices/{id}/servers/eligible`
`GET deliveryservices/{id}/servers`

Make sure you do not see the `asns` field in the response.

Run the same test for api version `4.1` and make sure that you see the `asns` field in the response this time.

Verify that all unit and API tests pass.
## If this is a bugfix, which Traffic Control versions contained the bug?

- master

## PR submission checklist
- [x] This PR has tests 
- [x] This PR has documentation 
- [ ] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

File: traffic_ops/traffic_ops_golang/deliveryservice/eligible.go
Comment (ocket8888) [inline traffic_ops/traffic_ops_golang/deliveryservice/eligible.go:99]: This causes APIv5 to return APIv4.0 structs instead of 4.1 - should check if API version is >= 4.1 instead of >=4.1 and < 5.0

**Review by ocket8888 — CHANGES_REQUESTED** [review]
Needs docs updates

File: traffic_ops/traffic_ops_golang/deliveryservice/eligible.go
Comment (srijeet0406) [reply traffic_ops/traffic_ops_golang/deliveryservice/eligible.go:99]: Ah yeah, forgot that we have API v5 now. I'll get that fixed.

File: traffic_ops/traffic_ops_golang/deliveryservice/servers/servers_test.go
Comment (zrhoffman) [inline traffic_ops/traffic_ops_golang/deliveryservice/servers/servers_test.go:203]: This part gets messy each time the struct version changes and has come up before (see https://github.com/apache/trafficcontrol/pull/4692#discussion_r426898413). If each field is assigned separately like
```go
	srv := tc.DSServerV4{}
	srv.ID = util.IntPtr(1)
	srv.Cachegroup = util.StrPtr("cgTest")
    // (and so on for the other fields)
	srv.ServerInterfaces = &[]tc.ServerInterfaceInfoV40{}
```
then that can be avoided.

File: traffic_ops/traffic_ops_golang/deliveryservice/eligible.go
Comment (srijeet0406) [reply traffic_ops/traffic_ops_golang/deliveryservice/eligible.go:99]: I see the `4.1` structs being returned for apiv5 calls. This check is for version 4.0 calls. All others(4.1 and 5.0) will default to 4.1. Are you seeing something different?

File: docs/source/api/v4/deliveryservices_id_servers.rst
Comment (zrhoffman) [inline docs/source/api/v4/deliveryservices_id_servers.rst:148]: Nit: Needs jq formatting

File: docs/source/api/v4/deliveryservices_id_servers_eligible.rst
Comment (zrhoffman) [inline docs/source/api/v4/deliveryservices_id_servers_eligible.rst:146]: Nit: Needs jq formatting

File: docs/source/api/v5/deliveryservices_id_servers.rst
Comment (zrhoffman) [inline docs/source/api/v5/deliveryservices_id_servers.rst:148]: Nit: Needs jq formatting

**Review by zrhoffman — CHANGES_REQUESTED** [review]

File: traffic_ops/traffic_ops_golang/deliveryservice/eligible.go
Comment (ocket8888) [reply traffic_ops/traffic_ops_golang/deliveryservice/eligible.go:99]: Oh, no my mistake, I simply read it wrong

File: traffic_ops/traffic_ops_golang/deliveryservice/servers/servers_test.go
Comment (zrhoffman) [reply traffic_ops/traffic_ops_golang/deliveryservice/servers/servers_test.go:204]: This was changed in `TestGetEligibleServers()` but not in `TestReadServers()`

**Review by zrhoffman — APPROVED** [review]
LGTM! I'll wait for @ocket8888 to approve before merging (unless he merges it first)

File: lib/go-tc/deliveryservice_servers.go
Comment (ocket8888) [inline lib/go-tc/deliveryservice_servers.go:209]: GoDoc comment should start with an optional article and the name of the documented symbol, and be a complete sentence.

File: docs/source/api/v5/deliveryservices_id_servers.rst
Comment (ocket8888) [inline docs/source/api/v5/deliveryservices_id_servers.rst:45]: This is native to APIv5, you don't need to let people know what minor version of a previous API added it

File: docs/source/api/v5/deliveryservices_id_servers_eligible.rst
Comment (ocket8888) [inline docs/source/api/v5/deliveryservices_id_servers_eligible.rst:51]: same as above

**Review by ocket8888 — CHANGES_REQUESTED** [review]
Docs changes all look good (comments are nitpicks), but changes still need to be made to the `/servers/{{ID}}` endpoint documentation

File: traffic_ops/traffic_ops_golang/server/servers.go
Comment (rimashah25) [inline traffic_ops/traffic_ops_golang/server/servers.go:1634]: Need to add `ORDER BY priority asc` in selectQuery in line 140.

**Review by rimashah25 — CHANGES_REQUESTED** [review]
Code changes to get specific version of server object with different version of API call looks good. Tested the changes on localhost with requested selectQuery change and the output was as expected.

File: lib/go-tc/deliveryservice_servers.go
Comment (ocket8888) [inline lib/go-tc/deliveryservice_servers.go:209]: GoDoc comments should begin with the name of the symbol they document

**Review by rimashah25 — APPROVED** [review]
Approving the changes for correct server struct response for different version of API calls.

**Review by ocket8888 — APPROVED** [review]

**Pull request merged**
